homeassistant/components/samsungtv/bridge.py

```diff
@@ -197,7 +197,7 @@ def __init__(self, method: str, host: str, port: int | None) -> None:
             CONF_TIMEOUT: 1,
         }
 
-    def mac_from_device(self) -> None:
+    def mac_from_device(self) -> str | None:
         """Try to fetch the mac address of the TV."""
         return None
```

Comment:
This is an abstract method and the type annotation should match the parent method. The child can choose to return any of the types that the parent has specified. It doesn't need to return all types to stay within spec.

Reply:
Ok, if the linters understand that, I agree it's better to use a more specific type annotation in the child while still staying within spec.

Reply:
I think in such case you don't have to handle all possible types. If you know that method only returns `None` then you can just not use the returned value at all.

I agree, there could be a case when you don't know that for sure and rely on IDE suggestions but I find such cases extremely rare and there this check can be ignored. In fact in the whole codebase I haven't found any place where it could be useful.

Reply:
Let's agree to disagree on this one